.buildbot.sh

```diff
@@ -0,0 +1,45 @@
+#! /bin/sh
+
+set -eu
+
+rm -fr ./build && mkdir ./build 
+cd build
+
+export CARGO_HOME="${PWD}/.cargo"
+export RUSTUP_HOME="${PWD}/.rustup"
+
+# install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup.sh
+sh rustup.sh --default-host x86_64-unknown-linux-gnu \
+    --default-toolchain nightly \
+    --no-modify-path \
+    --profile minimal \
+    -y
+export PATH=${CARGO_HOME}/bin/:$PATH
+
+rustup toolchain install nightly --allow-downgrade
+
+# yk bulid
+git clone --depth=1 https://github.com/ykjit/yk
+cd yk
+git submodule update --init --recursive
+cargo build
+cd ..
+
+# yklua bulid
+git clone --depth=1 https://github.com/ykjit/yklua
+
+export PATH=${PWD}/yk/bin:${PATH}
+export YK_BUILD_TYPE=debug
+
+cd yklua
+make
+cd ..
+
+cd .. # move to project root directory
+
+# yklua benchmark prob
+./build/yklua/src/lua ./yklua/fasta.lua 
```

Comment:
Should we be setting `YKD_SERIALISE_COMPILATION` here?

Maybe we should test with and without?

Reply:
Just seen that laurie asked you to remove it.

I disagree. I think we should test both with and without it set.

@ltratt ?

Reply:
Normal users will never set `YKD_SERIALISE_COMPILATION` so it shouldn't be part of the benchmark suite IMHO. [That doesn't mean that individuals can't test it, but as things stand I'd prefer not to have YKD_SERIALISE_COMPILATION percolate even more throughout our systems than we have already.]

Reply:
I see. I guess it depends on the purpose of `.buildbot.sh` in this repo.

In the past it's been used for correctness tests for shaking out bugs, but I think you are implying that in this instance that it's different.

I assume it won't be `.buildbot.sh` that actually collects our benchmark numbers?

Reply:
Oops, yes, I had overlooked that this was a buildbot line -- mea culpa!


But, that said, I still think we don't want to encourage more people to use `YKD_SERIALISE_COMPILATION` -- arguably it's a worse test than excluding it. We *could* include both, but I mostly view `YKD_SERIALISE_COMPILATION` as a hack, and I would prefer it doesn't leach to more repos, because people will end up assuming they should be using it too. IMHO, ykbench's job isn't to to check yk's correctness (if a benchmark here crashes yk, a test should end up in the yk repo).